File: xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 2/Python/double_click_test.py
from selenium import webdriver
from selenium.webdriver.common.action_chains import ActionChains
from array import *
import time, unittest

class DoubleClickTest (unittest.TestCase):
	def setUp(self)	:
		self.driver = webdriver.Chrome()
		
	def test_double_click(self):
		driver = self.driver
		driver.get("http://dl.dropbox.com/u/55228056/DoubleClickDemo.html")
		
		message = driver.find_element_by_id("message")
		
		#Verify color is Blue
		self.assertEqual("rgba(0, 0, 255, 1)", message.value_of_css_property("background-color"))
		
		ActionChains(self.driver).double_click(message).perform()
		
		#Verify Color is Yellow
		self.assertEqual("rgba(255, 255, 0, 1)", message.value_of_css_property("background-color"))
		
	def tearDown(self):
		self.driver.close()
		
if __name__ == "__main__":
	unittest.main()

File: xx_docs/selenium/1 - Selenium Testing Tools Cookbook [arquivo-apoio]/Chapter 2/Ruby/double_click_test.rb
require 'rubygems'
require 'selenium-webdriver'
require 'test/unit'

class DoubleClickTest < Test::Unit::TestCase
	def setup
		
		@driver = Selenium::WebDriver.for :chrome
		@driver.get 'http://dl.dropbox.com/u/55228056/DoubleClickDemo.html'
		@verification_errors = []
	end
	
	def test_double_click
		message = @driver.find_element :id => "message"
		
		#Verify color is Blue
		assert_equal "rgba(0, 0, 255, 1)", message.css_value('background-color').to_s()
		
		@driver.action.double_click(message).perform
		
		#Verify Color is Yellow
		assert_equal "rgba(255, 255, 0, 1)",message.css_value('background-color').to_s()
		
	end
	
	def teardown
		@driver.quit
		assert_equal [], @verification_errors
	end
end